openpype/modules/kitsu/utils/update_op_with_zou.py

```diff
@@ -260,15 +278,10 @@ def update_op_assets(
                 ancestor_id = None
 
         # Build OpenPype compatible name
-        if item_type in ["Shot", "Sequence"] and parent_zou_id is not None:
-            # Name with parents hierarchy "({episode}_){sequence}_{shot}"
-            # to avoid duplicate name issue
-            item_name = f"{item_data['parents'][-1]}_{item['name']}"
-
-            # Update doc name
-            asset_doc_ids[item["id"]]["name"] = item_name
-        else:
-            item_name = item["name"]
+        item_name = get_op_item_name_from_zou(
+            item, item_data['parents'][-1] if item_data["parents"] else None
+        )
+        asset_doc_ids[item["id"]]["name"] = item_name
```

Comment:
Wouldn't it technically make more sense to put the `parent` logic inside the `get_op_item_name_from_zou` function so that you just pass it the `item`?

Why do we need to have the separate `parent_name` argument?

Reply:
Just to be clear, I mean like this:
```python
def get_op_item_name_from_zou(
        item: dict
) -> str:
    """Helper function to build OpenPype item name from zou item.
    Args:
        item (dict): Zou item representation.
        parent_name (Optional[str]): Item parent's name or None.
    Returns:
          (str): Item name.
    """
    parent_name = item_data['parents'][-1] if item_data["parents"] else None
    if item["type"] in ["Shot", "Sequence"] and parent_name is not None:
        return f"{parent_name}_{item['name']}"
    return item["name"]
```

Reply:
A zou entity dict representation doesn't contain `parent` data but only a `parent_id`.  
`item_data` is a custom "item" representation with added data that are not present in a zou entity.

As I don't want to use custom representation in this function, I must stick to the zou entity representation.  
How to get the parent (and its name) from zou is not trivial and, to me, this intelligence should not be in this function.

And there are multiple ways to get the parent.  
For example in the lines you commented, the parent is get from the `parent_id`, and retrieved from all the project's assets, which can be a pretty slow process.  

In a custom add on I've on my side, it's retrieved from a zou db query, but only works from a zou task entity.  
Its much faster, but very specific to my needs.

That's why I prefere to let this intelligence outside of this function.  
Even if needing the parent_name as an argument is some kind of "single responsability" breaker.  

What about having the function accepting any `kwargs` ?  And use `parent_name` as a kwarg ?

Reply:
I missed that it used a separate variable, nvm.